### PR TITLE
feat(as): support lifecycle hook callback resource

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:
Support lifecycle hook callback resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
1. Support lifecycle hook callback resource.
2. Using environment variables to inject parameters for testing, because an instance with a status of **HANGING** cannot be obtained through resources and datasource.

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ export HW_AS_SCALING_GROUP_ID=xxxxxxxxxxxx
$ export HW_AS_LIFECYCLE_ACTION_KEY=xxxxxxxxxx
$ make testacc TEST="./huaweicloud/services/acceptance/as" TESTARGS="-run TestAccLifecycleHookCallBack_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccLifecycleHookCallBack_basic -timeout 360m -parallel 4
=== RUN   TestAccLifecycleHookCallBack_basic
=== PAUSE TestAccLifecycleHookCallBack_basic
=== CONT  TestAccLifecycleHookCallBack_basic
--- PASS: TestAccLifecycleHookCallBack_basic (6.54s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        6.585s



$ export HW_AS_SCALING_GROUP_ID=xxxxxxxxxxxx
$ export HW_AS_INSTANCE_ID=xxxxxxxxxx
$ export HW_AS_LIFECYCLE_HOOK_NAME=xxxxxxx
$ make testacc TEST="./huaweicloud/services/acceptance/as" TESTARGS="-run TestAccLifecycleHookCallBack_withInstanceIdAndHookName"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/as -v -run TestAccLifecycleHookCallBack_withInstanceIdAndHookName -timeout 360m -parallel 4
=== RUN   TestAccLifecycleHookCallBack_withInstanceIdAndHookName
=== PAUSE TestAccLifecycleHookCallBack_withInstanceIdAndHookName
=== CONT  TestAccLifecycleHookCallBack_withInstanceIdAndHookName
--- PASS: TestAccLifecycleHookCallBack_withInstanceIdAndHookName (6.26s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/as        6.360s

```

### CODE REVIEW

File: huaweicloud/services/as/resource_huaweicloud_as_lifecycle_hook_callback.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	callBackOpts := lifecyclehooks.CallBackOpts{
+		LifecycleActionResult: d.Get("lifecycle_action_result").(string),
+		LifecycleActionKey:    d.Get("lifecycle_action_key").(string),
+		InstanceId:            d.Get("instance_id").(string),
+		LifecycleHookName:     d.Get("lifecycle_hook_name").(string),
+	}
+
+	err = lifecyclehooks.CallBack(client, callBackOpts, d.Get("scaling_group_id").(string)).ExtractErr()
+	if err != nil {
+		return diag.Errorf("error creating AS lifecycle hook callback: %s", err)
```
Comment (Lance52259) [inline huaweicloud/services/as/resource_huaweicloud_as_lifecycle_hook_callback.go]:
```suggestion
		return diag.Errorf("unable to callback the AS lifecycle hook: %s", err)
```

Comment (jinyangyang222) [reply huaweicloud/services/as/resource_huaweicloud_as_lifecycle_hook_callback.go]: done.

**Pull request merged**